jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticator.java

```diff
@@ -103,6 +103,15 @@ public static Initializer initializer(String serverUrl, String repository) {
     return new Initializer(serverUrl, repository);
   }
 
+  @Nullable
+  private static Integer getHttpTimeout() {
+    Integer httpTimeout = Integer.getInteger("jib.httpTimeout");
+    if (httpTimeout != null && httpTimeout >= 0) {
```

Comment:
I think we could do just `return Integer.getInteger("jib.httpTimeout", null)` since in the negative case, we probably want to error (thrown by `HttpRequest#setConnectTimeout`)

Reply:
I was thinking showing a warning with a negative value, instead of erroring out. Do you think it's not good?

Anyways, if we decide to error out, I'll probably detect it and fail at the frontends with better error messages, rather than just displaying up an unfriendly stack trace from `Preconditions.checkArgument()` with no message.

Reply:
I think either way is good. I lean towards the errorring since the case I could think of where negative could show up is if the user may have meant to type another timeout and accidentally put a minus sign in front - in which case, a warning may be lost in the logs to a different error when the connection fails.

Reply:
I see. I think then the same argument applies to a non-parsable string values. The user could accidentally put any non-digit character, so it also makes sense to error out on non-parsable strings. Going for errorring on both cases?

Reply:
Yea, I think in the non-parsable version, this already errors, since `Integer#getInteger` would throw a `NumberFormatException`.

Reply:
Alright, errorring on both cases. (BTW, `Integer.getInteger()` doesn't throw `NumberFormatException`. `parseInt` does, OTOH.)

Reply:
Will update the code tomorrow, but I'm thinking of doing

```java
      Integer httpTimeout = null; // implies using the default timeout
      String timeoutValue = System.getProperty("jib.httpTimeout");
      if (timeoutValue != null) {
        httpTimeout = Integer.parseInt(timeoutValue);
        if (httpTimeout < 0) {
          throw new RegistryAuthenticationFailedException("Negative value of jib.httpTimeout");
        }
      }
      ...
      try (Connection connection = ...) {
        Request.Builder requestBuilder = Request.builder().setHttpTimeout(httpTimeout);
      ...
    } catch (NumberFormatException ex) {
      throw new RegistryAuthenticationFailedException("Non-integer value of jib.httpTimeout");
    }
```

Since all versions of `getInteger()` don't throw `NumberFormatException`, I guess I need to call `parseInt()` after `System.getProperty()`.

Reply:
@coollog and I agreed that we make the code simple and not care about the edge cases too much here. If something goes awry, so be it. Instead, we will guard the errors at the plugin frontends.